client/landing/gutenboarding/onboarding-block/style-preview/preview.tsx

```diff
@@ -11,13 +12,110 @@ import { STORE_KEY } from '../../stores/onboard';
 import * as T from './types';
 import { ValuesType } from 'utility-types';
 
+type Design = import('../../stores/onboard/types').Design;
+type SiteVertical = import('../../stores/onboard/types').SiteVertical;
+
 interface Props {
 	viewport: T.Viewport;
 	fonts: ValuesType< typeof import('../../constants').fontPairings >;
 }
 const Preview: React.FunctionComponent< Props > = ( { fonts, viewport } ) => {
-	const { selectedDesign } = useSelect( select => select( STORE_KEY ).getState() );
-	const [ { title: fontA }, { title: fontB } ] = fonts;
+	const [ previewHtml, setPreviewHtml ] = React.useState< string >();
+	const [ requestedFonts, setRequestedFonts ] = React.useState< Set< string > >( new Set() );
+
+	const { selectedDesign, siteVertical } = useSelect( select =>
+		select( STORE_KEY ).getState()
+	) as { selectedDesign: Design; siteVertical: SiteVertical };
+
+	const iframe = React.useRef< HTMLIFrameElement >( null );
+
+	React.useEffect(
+		() => {
+			const eff = async () => {
+				const [ { fontFamily: fontHeadings }, { fontFamily: fontBase } ] = fonts;
+				const templateUrl = `https://public-api.wordpress.com/rest/v1/template/demo/${ encodeURIComponent(
+					selectedDesign.slug
+				) }/${ encodeURIComponent( selectedDesign.slug ) }/`;
+				const url = addQueryArgs( templateUrl, {
+					language: 'en',
```

Comment:
Can we `useLangRouteParam` already? (Will the REST API endpoint fall back to `en`?)